handler/src/main/java/io/netty/handler/ssl/SslContext.java

```diff
@@ -93,6 +93,8 @@
         }
     }
 
+    final boolean startTls;
```

Comment:
Something like this should be alright I think in SniHandler (and then we can make the newHandler static method and the startTls member private in SslContext):
```java
        SslHandler sslHandler = null;
        try {
            sslHandler = sslContext.newHandler(ctx.alloc());
            ctx.pipeline().replace(this, SslHandler.class.getName(), sslHandler);
            sslHandler = null;
        } finally {
            // Since the SslHandler was not inserted into the pipeline the ownership of the SSLEngine was not
            // transferred to the SslHandler.
            // See https://github.com/netty/netty/issues/5678
            if (sslHandler != null) {
                ReferenceCountUtil.safeRelease(sslHandler.engine());
            }
        }
```